bin/propolis-server/src/lib/stats.rs

```diff
@@ -142,7 +138,9 @@ pub async fn start_oximeter_server(
         logging_config,
     };
 
-    let mut retry_print_timeout = 0;
+    const N_RETRY: u8 = 2;
+    const RETRY_WAIT_SEC: u64 = 1;
+    let mut attempts = 0;
```

Comment:
optional: I would consider rewriting this as `for attempts in 0..N_ATTEMPTS { ... }` and returning `Err` after the loop body. I think that makes it a little easier to reason about when to log the individual "attempt failed" messages and about the number of times the loop will try to connect (is it supposed to be 2 or 3? I think it ends up being 3 in this implementation, but I have to squint at it a bit to figure it out).

Reply:
I tried it that way initially and was mildly irritated at having to capture the last seen error somehow. I can change it back.

Reply:
Were you capturing the error just to return it? If the caller's discarding it now anyway (ILTM like it's just calling `.ok()` now), we could just have this routine return an Option instead of a Result.

Reply:
Yeah. We aren't using it in the caller, but it seemed wrong to me to just eat the error all together, since the logic of what to do with that error is a bit higher up. I don't feel that strongly since this isn't the long-term fix we want anyway.

Reply:
updated to use a for loop and return an Option in 2eb4619